mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/util/Classes.java

```diff
@@ -46,6 +52,9 @@ public Classes(Types typeUtils, Elements elementUtils) {
         typeUtils.erasure(elementUtils.getTypeElement(SettableByName.class.getName()).asType());
     this.gettableByNameType =
         typeUtils.erasure(elementUtils.getTypeElement(GettableByName.class.getName()).asType());
+    this.listElement = elementUtils.getTypeElement(List.class.getCanonicalName());
```

Comment:
That would add a significant amount of complexity because we'd need to make everything pluggable: how to create instances of this collection, how to add elements to it...

And that custom logic would need to be plugged into _generated_ code, which would make it even more complex.

We've never had that kind of request for mapper 3, so I don't think it's that common. And you always have the workaround of registering a custom codec to do the conversion.